### PR TITLE
Add micropython 1.18 and 1.19.1

### PR DESCRIPTION
**Pull request by dmitriy-serdyuk**
Make sure you have checked all steps below.

### Prerequisite
* [x] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
  * N/A
* [x] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
  * N/A
* [x] My PR addresses the following pyenv issue (if any)
  - Closes https://github.com/pyenv/pyenv/issues/2440

### Description
This commit will add fresh versions of microptyhon: 1.18 and 1.19.1.

### Tests
- [X] My PR adds the following unit tests (if any)
  * N/A

### CODE REVIEW

**Review by tomkins — CHANGES_REQUESTED** [review]
Frozen built in modules (eg. upip) won't work with this, due to: https://github.com/pyenv/pyenv/blob/a8afc611467d2a62f8d119205943395c1b5111de/plugins/python-build/bin/python-build#L902 and https://github.com/micropython/micropython/pull/8105 (and other commits/PRs)

pyenv MicroPython 1.19.1:

```
$ micropython
MicroPython v1.19.1 on 2022-08-29; darwin [GCC 4.2.1] version
Use Ctrl-D to exit, Ctrl-E for paste mode
>>> import sys
>>> sys.path
['', '/Users/tomkins/.pyenv/versions/micropython-1.19.1/lib/micropython']
>>>

$ micropython -m upip
ImportError: no module named 'upip'
```

Homebrew MicroPython 1.19.1:

```
$ /opt/homebrew/bin/micropython
MicroPython v1.19.1 on 2022-06-17; darwin [GCC 4.2.1] version
Use Ctrl-D to exit, Ctrl-E for paste mode
>>> import sys
>>> sys.path
['', '.frozen', '/Users/tomkins/.micropython/lib', '/usr/lib/micropython']
>>>

$ /opt/homebrew/bin/micropython -m upip
upip - Simple PyPI package manager for MicroPython
Usage: micropython -m upip install [-p <path>] <package>... | -r <requirements.txt>
import upip; upip.install(package_or_list, [<path>])

If <path> isn't given, packages will be installed to sys.path[1], or
sys.path[2] if the former is .frozen (path can be set from MICROPYPATH
environment variable if supported).
Default install path: /Users/tomkins/.micropython/lib

Note: only MicroPython packages (usually, named micropython-*) are supported
for installation, upip does not support arbitrary code in setup.py.
```

After tweaking `DMICROPY_PY_SYS_PATH_DEFAULT` in `python-build` to add `.frozen` to the path - upip worked fine.

**Comment by dmitriy-serdyuk** [comment]
Thanks! Added this change: now upip works. Just in case, I also tested older versions -- everything works.

File: plugins/python-build/bin/python-build [hunk 1]
```diff
@@ -899,7 +899,7 @@ build_package_micropython() {
     "$MAKE" $MAKE_OPTS
     cd ../ports/unix
     "$MAKE" $MAKE_OPTS axtls
-    "$MAKE" $MAKE_OPTS CFLAGS_EXTRA="-DMICROPY_PY_SYS_PATH_DEFAULT='\"${PREFIX_PATH}/lib/micropython\"' $CFLAGS_EXTRA"
+    "$MAKE" $MAKE_OPTS CFLAGS_EXTRA="-DMICROPY_PY_SYS_PATH_DEFAULT='\".frozen:${PREFIX_PATH}/lib/micropython\"' $CFLAGS_EXTRA"
```
Comment (native-api) [inline plugins/python-build/bin/python-build:902]: Won't this break previous versions?

If yes, the easiest way would probably be to set some kind of flag and do the corresponding thing depending on it. One way is to introduce another `build_*` function that would be setting it.

Comment (dmitriy-serdyuk) [reply plugins/python-build/bin/python-build:902]: I checked, it doesn't break previous versions:

```
$ micropython
MicroPython v1.17 on 2022-08-30; linux version
Use Ctrl-D to exit, Ctrl-E for paste mode
>>> import sys
>>> sys.path
['', '.frozen', '/home/serdyuk/.pyenv/versions/micropython-1.17/lib/micropython']
>>> import upip
>>>
```

**Comment by dmitriy-serdyuk** [comment]
Updated the test: now it passes on a local ubuntu machine.

**Comment by native-api** [comment]
The builds seem okay, the tests fail because MicroPython doesn't support `--version`. Is there an alternative?
E.g. does `-c 'import sys;print(sys.version)'` work for all releases?

**Pull request merged**